src/d/actor/d_a_obj_tower.cpp

```diff
@@ -4,50 +4,171 @@
 //
 
 #include "d/actor/d_a_obj_tower.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_procname.h"
+#include "dolphin/mtx/mtx.h"
 #include "dolphin/types.h"
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_printf.h"
 
 /* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    ((daObjTower_c*)i_this)->CreateHeap();
 }
 
 /* 00000098-0000020C       .text CreateHeap__12daObjTower_cFv */
-void daObjTower_c::CreateHeap() {
-    /* Nonmatching */
+bool daObjTower_c::CreateHeap() {
+    J3DModelData* pModel = (J3DModelData*)dRes_control_c::getRes(
+        "X_tower", 4, g_dComIfG_gameInfo.mResControl.mObjectInfo, 0x40);
+
+    if (pModel == NULL) {
+        JUTAssertion::showAssert(JUTAssertion::getSDevice(), "d_a_obj_tower.cpp", 0x56,
+                                 "modelData != 0");
+        OSPanic("d_a_obj_tower.cpp", 0x56, "Halt");
+    }
+
+    mpModel = mDoExt_J3DModel__create(pModel, 0, 0x11020203);
+
+    if (mpModel == NULL) {
+        return false;
+    } else {
+        PSMTXTrans(mDoMtx_stack_c::now, current.pos.x, current.pos.y, current.pos.z);
+        mDoMtx_YrotM(mDoMtx_stack_c::now, shape_angle.y);
+        mDoMtx_stack_c::scaleM(mScale.x, mScale.y, mScale.z);
+        PSMTXCopy(mDoMtx_stack_c::now, mMtx);
+
+        mpBgW = new dBgW();
+
+        if (mpBgW == NULL) {
+            return false;
+        } else {
+            cBgD_t* pData = (cBgD_t*)dRes_control_c::getRes(
+                "X_tower", 7, g_dComIfG_gameInfo.mResControl.mObjectInfo, 0x40);
+            return !mpBgW->Set(pData, cBgW::MOVE_BG_e, &mMtx);
+        }
+    }
 }
 
 /* 0000020C-0000028C       .text CreateInit__12daObjTower_cFv */
 void daObjTower_c::CreateInit() {
-    /* Nonmatching */
+    mCullMtx = mpModel->mBaseTransformMtx;
+    fopAcM_setCullSizeBox(this, -30000.0, -5000.0, -30000.0, 30000.0, 40000.0, 30000.0);
+    g_dComIfG_gameInfo.play.mBgS.Regist(mpBgW, this);
+    field_0x2d0 = 1;
+    set_mtx();
 }
 
 /* 0000028C-0000030C       .text set_mtx__12daObjTower_cFv */
 void daObjTower_c::set_mtx() {
-    /* Nonmatching */
+    J3DModel* pJVar1;
+    pJVar1 = mpModel;
+
+    (pJVar1->mBaseScale).x = mScale.x;
+    (pJVar1->mBaseScale).y = mScale.y;
+    (pJVar1->mBaseScale).z = mScale.z;
+
+    PSMTXTrans(mDoMtx_stack_c::now, current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_YrotM(mDoMtx_stack_c::now, current.angle.y);
+    PSMTXCopy(mDoMtx_stack_c::now, mpModel->mBaseTransformMtx);
 }
 
 /* 0000030C-000003DC       .text daObjTower_Create__FPv */
-static s32 daObjTower_Create(void*) {
-    /* Nonmatching */
+static cPhs__Step daObjTower_Create(void* i_this) {
+    cPhs__Step PVar3;
```

Comment:
As seen in the debug headers, daObjTower_Create calls an inline `daObjTower::_create`, which should help improve the casts here. The implementation of this method should just be:

```cpp
static s32 daObjTower_Create(void* i_this) {
    return ((daObjTower_c*)i_this)->_create();
}
```

with the rest of this method pulled into the _create method. This also goes for `_draw`, `_execute`, and `_delete`.